src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/client/VisitSchedulerClient.kt

```diff
@@ -0,0 +1,138 @@
+package uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.client
+
+import org.springframework.beans.factory.annotation.Qualifier
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.core.ParameterizedTypeReference
+import org.springframework.http.MediaType
+import org.springframework.stereotype.Component
+import org.springframework.web.reactive.function.BodyInserters
+import org.springframework.web.reactive.function.client.WebClient
+import org.springframework.web.reactive.function.client.bodyToMono
+import org.springframework.web.util.UriBuilder
+import uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.dto.ChangeVisitSlotRequestDto
+import uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.dto.ReserveVisitSlotDto
+import uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.dto.RestPage
+import uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.dto.SupportTypeDto
+import uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.dto.VisitCancelDto
+import uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.dto.VisitDto
+import uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.dto.VisitSessionDto
+import uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.filter.VisitSearchRequestFilter
+import java.time.Duration
+import java.util.Optional
+
+@Component
+class VisitSchedulerClient(
+
+  @Qualifier("visitSchedulerWebClient") private val webClient: WebClient,
+  @Value("\${visit-scheduler.api.timeout:10s}") val apiTimeout: Duration
+) {
+  private val pagedModel = object : ParameterizedTypeReference<RestPage<VisitDto>>() {}
```

Comment:
whats this for, is only being used once?

Reply:
As the visits by filter returns a paged response this object is used to create a Paged Response from the VSIP JSON 
response. It is only used on the visits by filter search.